k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -52,24 +52,29 @@ class EndpointsNamer(
   ): Activity[NameTree[Name]] = id.take(PrefixLen) match {
     case id@Path.Utf8(nsName, portName, serviceName) =>
       cache.services.flatMap { services =>
-        log.debug("k8s ns %s initial state: %s", nsName, services.keys.mkString(", "))
-        services.get(serviceName) match {
+        log.debug("k8s ns %s initial state: %s", nsName.toLowerCase, services.keys.mkString(", "))
+        services.get(serviceName.toLowerCase) match {
           case None =>
-            log.debug("k8s ns %s service %s missing", nsName, serviceName)
+            log.debug("k8s ns %s service %s missing", nsName.toLowerCase, serviceName.toLowerCase)
             Activity.value(NameTree.Neg)
 
-          case Some(services) =>
-            log.debug("k8s ns %s service %s found", nsName, serviceName)
-            services.ports.map { ports =>
-              ports.get(portName) match {
-                case None =>
-                  log.debug("k8s ns %s service %s port %s missing", nsName, serviceName, portName)
-                  NameTree.Neg
-
-                case Some(port) =>
-                  log.debug("k8s ns %s service %s port %s found + %s", nsName, serviceName, portName, residual.show)
-                  NameTree.Leaf(Name.Bound(port.addr, idPrefix ++ id, residual))
-              }
+          case Some(service) =>
+            log.debug("k8s ns %s service %s found", nsName.toLowerCase, serviceName.toLowerCase)
+            Try(portName.toInt).toOption match {
+              case Some(portNumber) =>
+                val addr = service.port(portNumber)
+                log.debug("k8s ns %s service %s port :%d found + %s", nsName.toLowerCase, serviceName.toLowerCase, portNumber, residual.show)
+                Activity.value(NameTree.Leaf(Name.Bound(addr, idPrefix ++ id, residual)))
+              case None =>
+                val state: Var[Activity.State[NameTree[Name.Bound]]] = service.port(portName.toLowerCase).map {
+                  case Some(addr) =>
+                    log.debug("k8s ns %s service %s port %s found + %s", nsName.toLowerCase, serviceName.toLowerCase, portName.toLowerCase, residual.show)
+                    Activity.Ok(NameTree.Leaf(Name.Bound(addr, idPrefix ++ id, residual)))
+                  case None =>
+                    log.debug("k8s ns %s service %s port %s missing", nsName.toLowerCase, serviceName.toLowerCase, portName.toLowerCase)
+                    Activity.Ok(NameTree.Neg)
+                }
+                Activity(state)
```

Comment:
my only concern here is that we seem to be calling toLowerCase _a lot_, which makes me worried that it's easy to miss this in one place..  Is there any way we can do this somewhere once than throughout the code?  For instance, can we just alter the Path once at the start?